### DISCUSSION TITLE
Lost event count functionality

### DISCUSSION THREAD

**Discussion by Engg-blogger** [bpftrace/bpftrace#3387] [General]
Hi Team - I have a question regarding the error "Lost <count> events". Could you explain me the functionality for this? The number of events that this message returns is real-time? or cumulative of past events? Another question, can you point me to list of other similar errors that could happen while tracing.

**Comment by ajor** [comment]
The lost events message shows the number of times your script has tried to put data onto a full ring buffer. The ring buffer is used for sending messages from BPF to userspace, mainly for `print`/`printf` statements. It's pretty-much real time - it shows the number of lost events since the last iteration of bpftrace's main ring buffer polling loop.

I don't think we've got a list of errors documented anywhere, but you can have bpftrace display all sorts of extra runtime warnings to you if you run with the `-kk` flag. Not all of these will actually be important though.
